.github/workflows/create-demo-registry.yml

```diff
@@ -0,0 +1,60 @@
+name: create-demo-registry
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 0/8 * * *"
+  pull_request:
+    branches:
+      - main
+      - sdk-preview
+    paths:
+      - cli/resources/registry/registry.yml
+      - infra/**
+      - .github/workflows/create-demo-registry.yml
+      - cli/setup.sh
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: check out repo
+      uses: actions/checkout@v2
+    - name: azure login
+      uses: azure/login@v1
+      with:
+        creds: ${{secrets.AZUREML_CREDENTIALS}}
+    - name: bootstrap resources
+      run: |
+          bash bootstrap.sh
+      working-directory: infra
+      continue-on-error: false
+    - name: setup-cli
+      run: |
+          source "${{ github.workspace }}/infra/sdk_helpers.sh";
+          source "${{ github.workspace }}/infra/init_environment.sh";
+          bash setup.sh
+      working-directory: cli
+      continue-on-error: true
+    - name: create registry
+      run: |
+          source "${{ github.workspace }}/infra/sdk_helpers.sh";
+          source "${{ github.workspace }}/infra/init_environment.sh";
+          az ml registry create -f resources/registry/registry.yml
+      working-directory: cli
+      continue-on-error: false
+    - name: validate registry
+      run: |
+          source "${{ github.workspace }}/infra/sdk_helpers.sh";
+          source "${{ github.workspace }}/infra/init_environment.sh";
+          az ml registry list --resource-group "${RESOURCE_GROUP_NAME}" --query "[?name == 'DemoRegistry']" |tail -n1|tr -d "[:cntrl:]"
+      working-directory: cli
+      continue-on-error: false
+    - name: delete registry
+      run: |
+          source "${{ github.workspace }}/infra/sdk_helpers.sh";
+          source "${{ github.workspace }}/infra/init_environment.sh";
+          az resource delete -g $RESOURCE_GROUP_NAME -n DemoRegistry --resource-type "Microsoft.MachineLearningServices/registries"
```

Comment:
Instead of custom workflows for registry, we can move this logic inside the shell scripts under cli directory and generate the workflows using python readme.py file. This way, we do not need to manage/maintain the registries workflows separately.